test/make.sh

```diff
@@ -74,11 +74,16 @@ function docker() {
 # directory paths which contain *.tf files.
 function check_terraform() {
   echo "Running terraform validate"
-  find_files . -name "*.tf" -print0 \
-    | compat_xargs -0 -n1 dirname \
-    | sort -u \
-    | grep -xv './test/fixtures/shared' \
-    | compat_xargs -t -n1 terraform validate --check-variables=false
+  find . -name "*.tf" \
+    -not -path "./.terraform/*" \
+    -not -path "./test/fixtures/*/.terraform/*" \
+    -not -path "./test/fixtures/all_examples/*" \
+    -not -path "./test/fixtures/shared/*" \
+    -print0 \
+    | xargs -0 dirname | sort | uniq \
+    | xargs -L 1 -i{} bash -c 'terraform init "{}" > /dev/null && terraform validate "{}"'
```

Comment:
Don't want to slow down the 0.12 activities, but just as an FYI this breaks the make checks on Mac OS X.

See https://github.com/terraform-google-modules/terraform-google-module-template/pull/28 for details and reasoning.

Suggest the following:

```bash
# find_files is a helper to exclude .git directories and match only regular
# files to avoid double-processing symlinks.
find_files() {
  local pth="$1"
  shift
  # Note: Take care to use -print or -print0 when using this function,
  # otherwise excluded directories will be included in the output.
  find "${pth}" '(' \
    -path '*/.git' -o \
    -path '*/.terraform' -o \
    -path '*/.kitchen' ')' \
    -prune -o -type f "$@"
}

# ...

function check_terraform() {
  set -e
  # fmt is before validate for faster feedback, validate requires terraform
  # init which takes time.
  echo "Running terraform fmt"
  find_files . -name "*.tf" -print0 \
    | compat_xargs -0 -n1 dirname \
    | sort -u \
    | compat_xargs -t -n1 terraform fmt -diff -check=true -write=false
  rval="$?"
  if [[ "${rval}" -gt 0 ]]; then
    echo "Error: terraform fmt failed with exit code ${rval}" >&2
    echo "Check the output for diffs and correct using terraform fmt <dir>" >&2
    return "${rval}"
  fi
  echo "Running terraform validate"
  # Change to a temporary directory to avoid re-initializing terraform init
  # over and over in the root of the repository.
  find_files . -name "*.tf" -print \
    | grep -v 'test/fixtures/shared' \
    | compat_xargs -n1 dirname \
    | sort -u \
    | compat_xargs -t -n1 test/terraform_validate
}
```

And finally, `test/terraform_validate` as

```bash
#! /bin/bash
#
# Copyright 2019 Google LLC. This software is provided as-is, without warranty
# or representation for any use or purpose. Your use of it is subject to your
# agreement with Google.
#
# This script initializes modules so that terraform validate as of 0.12 behaves
# as expected and does not issue errors such as:
#
# Error: Module not installed
#
#   on test/fixtures/shared_vpc_no_subnets/main.tf line 37:
#   37: module "project-factory" {
#
# This module is not yet installed. Run "terraform init" to install all modules
# required by this configuration.

# The first and only argument to this script is the directory containing *.tf
# files to validate.  This directory is assumed to be a root module.

set -eu
cd "${1}"
terraform init -backend=false >/dev/null
terraform validate
```

Reply:
We will resolve this next week.